controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java

```diff
@@ -1046,26 +1025,31 @@ private Slime toSlime(InputStream jsonStream) {
         }
     }
 
-    private void throwIfNotSuperUserOrPartOfOpsDbGroup(UserGroup userGroup, HttpRequest request) {
-        AthenzIdentity identity = authorizer.getIdentity(request);
-        if (!(identity instanceof AthenzUser)) {
-            throw new ForbiddenException("Identity not an user: " + identity.getFullName());
-        }
-        AthenzUser user = (AthenzUser) identity;
-        if (!authorizer.isSuperUser(request) && !authorizer.isGroupMember(new UserId(user.getName()), userGroup) ) {
-            throw new ForbiddenException(String.format("User '%s' is not super user or part of the OpsDB user group '%s'",
-                                                       user.getName(), userGroup.id()));
-        }
-    }
-
     private void throwIfNotAthenzDomainAdmin(AthenzDomain tenantDomain, HttpRequest request) {
-        AthenzIdentity identity = authorizer.getIdentity(request);
-        if ( ! authorizer.isAthenzDomainAdmin(identity, tenantDomain)) {
+        AthenzIdentity identity = getUserPrincipal(request).getIdentity();
+        boolean isDomainAdmin = athenzClientFactory.createZmsClientWithServicePrincipal()
+                .isDomainAdmin(identity, tenantDomain);
+        if ( ! isDomainAdmin) {
             throw new ForbiddenException(
                     String.format("The user '%s' is not admin in Athenz domain '%s'", identity.getFullName(), tenantDomain.getName()));
         }
     }
 
+    private static Optional<UserId> getUserId(HttpRequest request) {
+        return Optional.of(getUserPrincipal(request))
+                .map(AthenzPrincipal::getIdentity)
+                .filter(AthenzUser.class::isInstance)
+                .map(AthenzUser.class::cast)
+                .map(AthenzUser::getName)
+                .map(UserId::new);
+    }
+
+    private static AthenzPrincipal getUserPrincipal(HttpRequest request) {
+        return Optional.ofNullable(request.getJDiscRequest().getUserPrincipal())
+                .map(AthenzPrincipal.class::cast)
```

Comment:
Missing `isInstance` check before cast.

Reply:
It should fail hard if it's not an `AthenzPrincipal`. I agree though that `ClassCastException` is not the best way of doing that...